qiskit_nature/second_q/properties/grouped_property.py

```diff
@@ -15,14 +15,15 @@
 from __future__ import annotations
 
 from collections.abc import Iterable
-from typing import Generator, Generic, Optional, Type, TypeVar, Union
+from typing import Generator, Generic, Optional, Type, TypeVar, Union, TYPE_CHECKING
 
 import h5py
-
 from qiskit_nature.hdf5 import _import_and_build_from_hdf5
-from qiskit_nature.results import EigenstateResult
+
 from .property import Interpretable, Property
 
+if TYPE_CHECKING:
+    from qiskit_nature.second_q.problems import EigenstateResult
```

Comment:
I am assuming that there are cyclic import issues or something that where this is the "fix". Hopefully we can sort it later so that we do not have to resort to making the types strings.